File: ios/LoadcastCore/Sources/LoadcastCore/LoadCalculator.swift
public enum LoadCalculator {

    /// 부하(점) = 운동 시간(분) × 강도 계수. docs/SPEC.md §2.
    public static func sessionLoad(durationMin: Int, intensity: Intensity) -> Int {
        durationMin * intensity.coefficient
    }

    /// [from, to] (양끝 포함) 윈도우의 부하 합.
    public static func totalLoad(records: [WorkoutRecord], from: DayDate, to: DayDate) -> Int {
        records
            .filter { $0.date >= from && $0.date <= to }
            .reduce(0) { $0 + $1.load }
    }
}

File: android/core/src/main/kotlin/com/loadcast/core/LoadCalculator.kt
package com.loadcast.core

import java.time.LocalDate

object LoadCalculator {

    /** 부하(점) = 운동 시간(분) × 강도 계수. docs/SPEC.md §2. */
    fun sessionLoad(durationMin: Int, intensity: Intensity): Int =
        durationMin * intensity.coefficient

    /** [from, to] (양끝 포함) 윈도우의 부하 합. */
    fun totalLoad(records: List<WorkoutRecord>, from: LocalDate, to: LocalDate): Int =
        records.filter { it.date in from..to }.sumOf { it.load }
}
